connector/src/main/scala/com/basho/riak/spark/writer/RiakWriter.scala

```diff
@@ -97,44 +99,70 @@ class RiakTSWriter[T](connector: RiakConnector,
                       writeConf: WriteConf) extends RiakWriter[T, RowDef](connector, bucketDef, dataMapper, writeConf) {
   override def store(session: RiakSession, ns: Namespace, objects: Iterator[T],
                      dataMapper: WriteDataMapper[T, RowDef], writeConf: WriteConf): Unit = {
-    val rowDefs = objects.map(dataMapper.mapValue).toList
-
-    val builder = new StoreOperation.Builder(ns.getBucketNameAsString).withRows(rowDefs.map(_.row))
-    rowDefs.map(_.columnDescription).filter(_.isDefined).map(_.get).headOption.foreach(descr => builder.withColumns(descr.toList))
-
-    val future: RiakFuture[Void, String] = session.execute(builder.build())
-    future.await()
+    @volatile var riakException: Throwable = null
```

Comment:
It's better to use Option instead of initialising with null. And later we can use Option#foreach() method instead of if (... != null) to check if exception occurred